File: airbyte-integrations/connectors/destination-redshift-py/destination_redshift_py/field.py
class DataType:
    def __init__(self, name: str, length: str = None):
        self.name = name
        self.length = length

    def __str__(self):
        if self.length:
            return f"{self.name}({self.length})"
        else:
            return self.name


class Field:
    def __init__(self, name: str, data_type: DataType, value: object = None):
        self.name = name
        self.data_type = data_type
        self.value = value

    def __str__(self):
        return f""""{self.name}" {self.data_type}"""

File: airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/Field.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.protocol.models;

import java.util.List;

public class Field extends CommonField<JsonSchemaType> {

  private List<Field> subFields;

  public Field(final String name, final JsonSchemaType type) {
    super(name, type);
  }

  public Field(final String name, final JsonSchemaType type, List<Field> subFields) {
    super(name, type);
    this.subFields = subFields;
  }

  public static Field of(final String name, final JsonSchemaType type) {
    return new Field(name, type);
  }

  public static Field of(final String name, final JsonSchemaType type, List<Field> properties) {
    return new Field(name, type, properties);
  }

  public List<Field> getSubFields() {
    return subFields;
  }

}
